packages/vm/src/evm/message.ts

```diff
@@ -1,48 +1,84 @@
 import { Address } from 'ethereumjs-util'
 import { PrecompileFunc } from './precompiles'
 
+const defaults = {
+  value: BigInt(0),
+  caller: Address.zero(),
+  data: Buffer.alloc(0),
+  depth: 0,
+  isStatic: false,
+  isCompiled: false,
+  delegatecall: false,
+}
+
+interface MessageOpts {
+  to?: Address
+  value?: bigint
+  caller?: Address
+  gasLimit: bigint
+  data?: Buffer
+  depth?: number
+  code?: Buffer | PrecompileFunc
+  codeAddress?: Address
+  isStatic?: boolean
+  isCompiled?: boolean
+  salt?: Buffer
+  selfdestruct?: { [key: string]: boolean } | { [key: string]: Buffer }
+  delegatecall?: boolean
+  authcallOrigin?: Address
+}
+
 export default class Message {
-  to: Address
+  to?: Address
   value: bigint
   caller: Address
   gasLimit: bigint
   data: Buffer
   depth: number
-  code: Buffer | PrecompileFunc
-  _codeAddress: Address
+  code?: Buffer | PrecompileFunc
+  _codeAddress?: Address
   isStatic: boolean
   isCompiled: boolean
-  salt: Buffer
-  selfdestruct: any
+  salt?: Buffer
+  selfdestruct?: { [key: string]: boolean } | { [key: string]: Buffer }
```

Comment:
can you add a typedoc with your findings/understanding of the two values (boolean vs buffer), i.e. set to boolean when marked for selfdestruct and set to buffer for the refund address

Reply:
Added a note! :)